UnitTests/Views/ColorPicker2Tests.cs

```diff
@@ -216,6 +216,75 @@ public void ColorPicker_RGB_NoText (Color c, string expected)
         cp.Draw ();
 
 
+        TestHelpers.AssertDriverContentsAre (expected, output);
+
+        top.Dispose ();
+    }
+
+    public static IEnumerable<object []> ColorPickerTestData_WithTextFields()
+    {
+        yield return new object []
+        {
+            new Color(255, 0, 0),
+            @"
+R:█████████████▲255 
```

Comment:
I've really grown to despise unit tests that test for visuals. They are extremely hard to maintain over the long run.

Can these tests be replaced with something that either doesn't test contents or is easier to change if elements are updated?

Reply:
We can make the triangle location from a local variable to a public `get` for the test

And we can replace the diagram code with some test helpers e.g. https://github.com/gui-cs/Terminal.Gui/pull/3604/commits/d4697bdca002e4102ca2d6c70172ee8446a1e9aa

I don't think it will make it more maintainable though mostly just make it harder to figure out what is going on when it does break.

Reply:
Tests now take the format:

```csharp
    [Fact]
    [AutoInitShutdown]
    public void ColorPicker_Construct_DefaultValue ()
    {
        var cp = new ColorPicker2 { Width = 20, Height = 4, Value = new (0, 0) };

        cp.Style.ShowTextFields = false;
        cp.ApplyStyleChanges ();

        var top = new Toplevel ();
        top.Add (cp);
        Application.Begin (top);

        // Should be only a single text field (Hex) because ShowTextFields is false
        Assert.Single (cp.Subviews.OfType<TextField> ());

        cp.Draw ();

        // All bars should be at 0 with the triangle at 0 (+2 because of "H:", "S:" etc)
        var h = GetColorBar (cp, ColorPickerPart.Bar1);
        Assert.Equal ("H:", h.Text);
        Assert.Equal (2, h.TrianglePosition);
        Assert.IsType<HueBar> (h);

        var s = GetColorBar (cp, ColorPickerPart.Bar2);
        Assert.Equal ("S:", s.Text);
        Assert.Equal (2, s.TrianglePosition);
        Assert.IsType<SaturationBar> (s);

        var v = GetColorBar (cp, ColorPickerPart.Bar3);
        Assert.Equal ("V:", v.Text);
        Assert.Equal (2, v.TrianglePosition);
        Assert.IsType<ValueBar> (v);

        var hex = GetTextField (cp, ColorPickerPart.Hex);
        Assert.Equal ("#000000", hex.Text);

        top.Dispose ();
    }
```

Reply:
Which one refers to color?

Reply:
> Which one refers to color?

Not sure I understand the question.  Can you elaborate?

Reply:
I think I confused GetColorBar with GetColor. I'm sorry.